### DISCUSSION TITLE
Is it possible to return responses from `asyncio.as_completed` as they are completed.

### DISCUSSION THREAD

**Discussion by MichaelJThomas-2016** [fastapi/fastapi#10103] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import asyncio
import datetime
from typing import List

from fastapi import APIRouter, Depends

from db.session import async_session
from llm import get_query_engine
from util import _get_date_range
from models import LLMSummaryInput

router = APIRouter()


async def generate_response(input_: LLMSummaryInput):
    engine = await get_query_engine(
        input_   
    )
    qe = await engine["query_engine"].aquery(llm_input.prompt)
    summary = qe.response
    return summary

async def process_one_day(date, exsists: dict):
    obj_exists = exsists.get(date)
    if obj_exists:  # Some cached/DB stuff
        return obj_exists

    else:
        res = await generate_response(date)
        return res


@app.get("/my_objs", dependencies=[Depends(check_auth_token)])
def my_objs(date_range: int):
    dates: List[datetime.date] = _get_date_range(date_range=date_range)  # pd.date_range...
    async with async_session() as session:
        db_objs = await crud_summary.get_aysnc_multi_by_day(
            db=session,
            start_date=dates[0],
            end_date=dates[-1]
        )
    obj_map = {x.date: x for x in db_objs}

    async def stream_responses():
        tasks = []
        for date in dates:
            tasks.append(
                asyncio.create_task(process_one_day(
                    date=date,
                ))
            )
        for response in asyncio.as_completed(tasks):
            await response
            yield response.result()

    return StreamingResponse(
        content=stream_responses()
        ...
    )


```

### Description

The code does not exactly encompass my entire issue, but I am trying to see if it is possible to  send responses as they are processed in the asyncio.as_completed loop.

My javascript implements the [ReadableStreams](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream), but it appears as though the responses are only sent after all the tasks have completed, though i can see them finishing in my logger.

I've see references to `run_in_threadpool`, but it doesn't seem to help.

any help would be appreciated.

### Operating System

macOS

### FastAPI Version

0.99.1

### Pydantic Version

1.10.12

### Python Version

3.11

### Additional Context

<i> Yes... it's another LLM thing </i>

**Comment by Kludex** [comment]
If you can provide a minimal example that I don't need to install anything, I can help.

**Reply by MichaelJThomas-2016** [reply]
@Kludex - Thank you. I'll get back to you with an example.
